pkgs/development/tools/yarn/default.nix

```diff
@@ -1,26 +0,0 @@
-{ stdenv, nodejs, fetchzip, makeWrapper }:
-
-stdenv.mkDerivation rec {
-  name = "yarn-${version}";
-  version = "1.6.0";
-
-  src = fetchzip {
-    url = "https://github.com/yarnpkg/yarn/releases/download/v${version}/yarn-v${version}.tar.gz";
-    sha256 = "0bblp1jy4s9y5rpcqn40w61qwsmxr342xkcn7ykk88i7sng2cgfw";
-  };
-
-  buildInputs = [makeWrapper nodejs];
-
-  installPhase = ''
-    mkdir -p $out/{bin,libexec/yarn/}
-    cp -R . $out/libexec/yarn
-    makeWrapper $out/libexec/yarn/bin/yarn.js $out/bin/yarn
```

Comment:
Could your problem be also solved by adding a `yarnpkg` symlink?

Reply:
This `makeWrapper` call also does not work on macOS because shebangs executables are not allowed to be scripts itself. It should use the nodejs binary instead for wrapping.